Lib/test/test_codecs.py

```diff
@@ -1641,6 +1641,13 @@ def test_register(self):
         self.assertRaises(TypeError, codecs.register)
         self.assertRaises(TypeError, codecs.register, 42)
 
+    def test_unregister(self):
+        search_function = mock.Mock(return_value=(1, 2, 3, 4))
+        codecs.register(search_function)
```

Comment:
Hum, could you check that here it is called? Something like:

```
        name = "nonexistent_codec_name"
        ...
        self.assertRaises(LookupError, codecs.lookup, name)
        search_function.assert_called_with(name)
        search_function.reset_mock()

        codecs.unregister(search_function) 
        ...
```

Reply:
How about this new test case?

Reply:
Yep, perfect. I better trust the test written like this.